pkg/ebpf/ingress_node_firewall_events.go

```diff
@@ -57,12 +66,70 @@ func (infc *IngNodeFwController) ingressNodeFwEvents() error {
 				continue
 			}
 
+			// read the event header
+			if _, err := io.ReadFull(bytes.NewBuffer(record.RawSample), buf[:]); err != nil {
+				log.Printf("parsing perf event header err: %s", err)
+				continue
+			}
+			// Note position of the bytes in the buf slice depends on the layout of bpfEventHdrSt struct
+			eventHdr.RuleId = binary.LittleEndian.Uint16(buf[2:4])
+			eventHdr.Action = buf[4]
+			eventHdr.PktLength = binary.LittleEndian.Uint16(buf[6:8])
+			packet := make([]byte, eventHdr.PktLength)
 			// Parse the perf event entry into a bpfEvent structure.
-			if err := binary.Read(bytes.NewBuffer(record.RawSample), binary.LittleEndian, &event); err != nil {
-				log.Printf("parsing perf event: %s", err)
+			if err := binary.Read(bytes.NewBuffer(record.RawSample[eventHdrSize:]), binary.LittleEndian, &packet); err != nil {
+				log.Printf("parsing perf event packet header : %s", err)
 				continue
 			}
+			log.Printf("========= Drop Event ===========")
```

Comment:
Will this log every dropped packet? I think for now, that level of verbosity is fine, but for production environments, we should only log the first drop for a given flow or not at all

Reply:
we can have it debug level or something we can't know 1st pkt from flow or not